frontend/src/metabase/public/components/widgets/SharingPane/SharingPane.unit.spec.tsx

```diff
@@ -0,0 +1,194 @@
+        ).toBeInTheDocument();
+      });
+
+      it("should call `onChangeEmbedType` with `application` when `Edit settings` is clicked", () => {
+        const { onChangeEmbedType } = setup({ isResourcePublished: true });
+
+        userEvent.click(screen.getByRole("button", { name: "Edit settings" }));
+
+        expect(onChangeEmbedType).toHaveBeenCalledWith("application");
+      });
+    });
+
+    describe("when the resource is not published", () => {
+      it("should render `Set this up`", () => {
+        setup({ isResourcePublished: false });
+
+        expect(
+          screen.getByRole("button", { name: "Set this up" }),
+        ).toBeInTheDocument();
+      });
+
+      it("should call `onChangeEmbedType` with `application` when `Set this up` is clicked", () => {
+        const { onChangeEmbedType } = setup({ isResourcePublished: false });
+
+        userEvent.click(screen.getByRole("button", { name: "Set this up" }));
+
+        expect(onChangeEmbedType).toHaveBeenCalledWith("application");
+      });
+    });
+  });
+  describe("public embed button", () => {
+    describe("when public sharing is disabled", () => {
+      it("should render link to settings and a disabled button with `Get an embed link`", () => {
+        setup({ isPublicSharingEnabled: false });
+
+        expect(
+          screen.getByText("Public embeds and links are disabled."),
+        ).toBeInTheDocument();
+        expect(
+          screen.getByTestId("sharing-pane-settings-link"),
+        ).toBeInTheDocument();
+
+        const embedLinkButton = screen.getByRole("button", {
+          name: "Get an embed link",
+        });
+        expect(embedLinkButton).toBeInTheDocument();
+        expect(embedLinkButton).toBeDisabled();
+      });
+
+      it("should redirect to settings when public sharing is disabled and `Settings` is clicked", () => {
+        const { history } = setup({
+          isPublicSharingEnabled: false,
+        });
+
+        userEvent.click(screen.getByTestId("sharing-pane-settings-link"));
+
+        expect(history.getCurrentLocation().pathname).toEqual(
+          "/admin/settings/public-sharing",
+        );
+      });
+    });
+
+    describe("when a public link exists", () => {
+      it("should render iframe link, copy button, and `Copy snippet` description", () => {
+        setup({ hasPublicLink: true, isPublicSharingEnabled: true });
+
+        expect(
+          screen.getByText(
+            "Just copy this snippet to add a publicly-visible iframe embed to your web page or blog post.",
+          ),
+        ).toBeInTheDocument();
+
+        expect(screen.getByTestId("public-link-text")).toBeInTheDocument();
+        expect(screen.getByTestId("public-link-text")).toHaveTextContent(
+          /<iframe src="mock-uuid".*<\/iframe>/i,
+        );
+
+        expect(screen.getByTestId("copy-button")).toBeInTheDocument();
+        expect(screen.getByText("Remove public URL")).toBeInTheDocument();
```

Comment:
should we also test that `"Affects both embed URL and public link for this dashboard"` is rendered when hover over the remove button?

Reply:
yep

Reply:
added